src/main/kotlin/LinkedList.kt

```diff
@@ -10,18 +10,58 @@ class LinkedList<T> {
     
     fun count(): Int = size
        
-    fun insert(index: Int, obj: T) {
-        TODO("Not yet implemented")
-    }
-    fun removeAt(index: Int) {
-        TODO("Not yet implemented")
+    private fun getNode(index: Int): Node<T> {
+        var current = head
+        var currentIndex = 0
+        while (current != null && currentIndex < index) {
+            current = current.nextNode
+            currentIndex++
+        }
+        return current!!
     }
+    
     operator fun get(index: Int): T {
-        TODO("Not yet implemented")
+        require(index in 0 until size-1)
+        return getNode(index).value
     }
+    
     operator fun set(index: Int, obj: T) {
         TODO("Not yet implemented")
     }
+    
+    fun insertFirst(value: T) {
+        head = Node(value = value, nextNode = head)
+        if (last == null) {
+            last = head
+        }
+        size++
+    }
+
+    fun insertLast(value: T) {
+        val newLast = Node(value = value)
+        last?.nextNode = newLast
+        last = newLast
+        if (size == 0) head = newLast
+        size++
+    }
+    
+    fun insert(index: Int, value: T) {
+        require(index in 0 until size)
+        if (size==0 || index == size) {
+            insertLast(value)
+        }
+        else {
+            var prevNode = getNode(index-1)
+            var newNode = Node(value, prevNode.nextNode)
+            prevNode.nextNode = newNode
+            size++
+        }
```

Comment:
Я всё ещё не понимаю два момента:
1. Зачем условие $\mathrm{size} = 0$, если оно &ndash; частный случай $\mathrm{index} = \mathrm{size}$?
1. В ситуации $\mathrm{size} > 0$, $\mathrm{index} = 0$, забывается смена `head`. Да и вообще `getNode(index - 1)` некорректно работает, так как $\mathrm{index} - 1 = -1 < 0$.

Я предлагаю сначала проверить начало, затем, конец, затем всё остальное:
```suggestion
        when (index) {
            0 -> insertFirst(value)
            size -> insertLast(value)
            else -> {
                var prevNode = getNode(index-1)
                var newNode = Node(value, prevNode.nextNode)
                prevNode.nextNode = newNode
                size++
            }
        }
```

Reply:
да, после изменения insertLast стало работать некорректно